authentication/account/service/user.go

```diff
@@ -98,6 +100,50 @@ func (s *userServiceImpl) DeprovisionUser(ctx context.Context, username string)
 	return identity, err
 }
 
+// DeactivateUser deactivates a user, i.e., mark her as `active=false`, obfuscate the personal info and soft-delete the account
+func (s *userServiceImpl) DeactivateUser(ctx context.Context, username string) (*repository.Identity, error) {
+
+	var identity *repository.Identity
+	err := s.ExecuteInTransaction(func() error {
+
+		identities, err := s.Repositories().Identities().Query(
+			repository.IdentityWithUser(),
+			repository.IdentityFilterByUsername(username),
+			repository.IdentityFilterByProviderType(repository.DefaultIDP))
+		if err != nil {
+			return err
+		}
+		if len(identities) == 0 {
+			return errors.NewNotFoundErrorWithKey("user identity", "username", username)
+		}
+
+		identity = &identities[0]
+		// mark the account as inactive
+		identity.User.Active = false
+		// obfuscate the data
+		obfuscatated := uuid.NewV4().String()
+		identity.Username = obfuscatated
+		identity.ProfileURL = &obfuscatated
+		identity.User.Email = obfuscatated
+		identity.User.FullName = obfuscatated
+		identity.User.ImageURL = obfuscatated
+		identity.User.URL = obfuscatated
+		identity.User.Company = obfuscatated
+		identity.User.Bio = obfuscatated
+		identity.User.Cluster = obfuscatated
+		identity.User.FeatureLevel = obfuscatated
+		// empty data
+		identity.User.ContextInformation = account.ContextInformation{}
+		err = s.Repositories().Identities().Save(ctx, identity)
+		if err != nil {
+			return err
+		}
+		return s.Repositories().Users().Save(ctx, &identity.User)
```

Comment:
You do obfuscate the data but I don't see you soft delete User & Identity.

Also load after obfuscating and deleting User and Identity from the default IdP we would want to load all other Identities (not in use currently, but still theoretically can be used in the future) and obfuscate&soft-delete them too. But it's low priority and can be handled in a separate issue/PR in the future.

Reply:
soft-delete will be taken care of in https://github.com/fabric8-services/fabric8-auth/issues/778 which is also part of https://github.com/fabric8-services/fabric8-auth/issues/774, but I thought it would be simpler to review the work if it was splitted in small subtasks

Reply:
OK. That looks fine to me.